PatientAnalytics/Services/DatabasePopulateService.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.EntityFrameworkCore;
+using PatientAnalytics.Models;
+using PatientAnalytics.Models.Auth;
+using PatientAnalytics.Utils;
+
+namespace PatientAnalytics.Services;
+
+public class DatabasePopulateService(IServiceScopeFactory scopeFactory, IConfiguration configuration) : IHostedService
+{
+    public async Task StartAsync(CancellationToken stoppingToken)
+    {
+        using var scope = scopeFactory.CreateScope();
+
+        await using var context = scope.ServiceProvider.GetRequiredService<Context>();
+
+        await context.Database.EnsureCreatedAsync(stoppingToken);
+
+        var isMigrationNeeded = (await context.Database.GetPendingMigrationsAsync(stoppingToken)).Any();
+
+        if (isMigrationNeeded)
+        {
+            await context.Database.MigrateAsync(stoppingToken);
+        }
+
+        if (!context.Users.Any(u => u.Role == "SuperAdmin"))
```

Comment:
Two suggestions with this one:

1. is it worth giving your super admin a static user id and searching on that? This could end up scanning the whole table, which could get large in future
2. Try and use the async overload where possible

Reply:
In that case should it check whether User with ID: 1 exist? Since it will be the first user and the intention is to populate a SuperAdmin on initial run of this application locally.

Should I use `AnyAsync` instead, as part of trying to use async overload whenever possible?